File: app/src/main/java/com/moeiny/reza/dmrc_booking/Node.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.moeiny.reza.dmrc_booking;

import androidx.annotation.NonNull;

import org.json.JSONException;
import org.json.JSONObject;

public class Node {

    public long id;
    public String name;
    public boolean isExchange;

    public Node(long id, String name) {
        this.id = id;
        this.name = name;
    }

    public Node(JSONObject station) throws JSONException {
        this.id = station.getLong("id");
        this.name = station.getString("name");
        this.isExchange = station.getBoolean("isExchange");
    }

    @NonNull
    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/com/moeiny/reza/dmrc_booking/model/Node.kt
package com.moeiny.reza.dmrc_booking.model

import org.json.JSONException
import org.json.JSONObject

class Node(station: JSONObject) {
    var id: Long = 0
    var name: String? = null
    var isExchange = false


    init {
        id = station.getLong("id")
        name = station.getString("name")
        isExchange = station.getBoolean("isExchange")
    }

   override fun toString(): String {
        return name!!
    }
}